api/gateway-operator/v1beta1/gatewayconfiguration_types.go

```diff
@@ -62,6 +64,16 @@ type GatewayConfigurationSpec struct {
 	// +optional
 	ControlPlaneOptions *ControlPlaneOptions `json:"controlPlaneOptions,omitempty"`
 
+	// ListenerOptions is the specification for configuration bound to specific listeners in the Gateway.
+	// It will override the default configuration of control plane or data plane for the specified listener.
+	// +kubebuilder:validation:MaxItems=64
+	//
+	// +optional
+	// +kubebuilder:validation:MaxItems=64
```

Comment:
REVIEW: Should we keep the same number of `MaxItems` with upstream here? Seems that we do not support 64 ports for ingress service currently.